### DISCUSSION TITLE
pushgateway does not support HTTPS

### DISCUSSION THREAD

**Discussion by aderocha** [prometheus/pushgateway#505] [Q&A]
Hi,
we have a pushgateway that only accepts HTTPS rquests and it was a big surprise when we realized that pushgateway class only accepts HTTP and is hard-coded right in the constructor:

  public PushGateway(String address) {
    this(createURLSneakily("http://" + address));
  }

I'm using latest version available for Java (0.16.0).

I have 2 questions:
Why does this limitation exists ?
and
Can we expect pushgateway to support other protocols like HTTPS in the near future ?

**Comment by beorn7** [comment]
The Pushgateway does support TLS, see https://github.com/prometheus/pushgateway#tls-and-basic-authentication

**Comment by beorn7** [comment]
What you are referring to is the client side. I suggest to ask over in https://github.com/prometheus/client_java ;)
